### DISCUSSION TITLE
PYZOR and RAZOR weight is always 0

### DISCUSSION THREAD

**Discussion by TomFuch** [rspamd/rspamd#3996] [Q&A] [answered]
I installed PYZOR and RAZOR as described here: https://rspamd.com/doc/modules/external_services.html#pyzor-specific-details

All is working as expected, but the Weight of PYZOR and RAZOR is always 0.
Is it necessary to change the default weight?

This ist the entry in Web UI: **PYZOR (0) [bl_17_wl_0]**

Thanks for hints...

**Comment by segmentierungsfehler** [comment, marked as answer]
Here is my local.d/external_services.conf
```
pyzor {
  symbol="PYZOR"
  servers = "127.0.0.1:5953"
  timeout = 10.0
  cache_expire = 7200
  retransmits = 1
 }
razor {
  symbol="RAZOR"
  servers = "127.0.0.1:11342"
  timeout = 10.0
  cache_expire = 7200
}
```
and my local.d/external_services_group.conf
```
symbols {
    "RAZOR" {
        weight = 3.0;
        description = "check message digest on razor"
    }
    "PYZOR" {
        weight = 3.5;
        description = "check message digest on pyzor"
    }
}
```
This config worked for me with rspamd version 3.0 and 3.1.

**Reply by TomFuch** [reply]
Thanks! I had nearly the same settings, except RAZOR weight to 5 and PYZOR weight to 3. Why do u change timeout and retransmits from the default?

**Reply by segmentierungsfehler** [reply]
During my tests i found out, that razor or pyzor checks never took longer than 5 seconds. So i reduced the default timeout of 15 to 10 seconds. I also wanted to avoid to many retransmits during the tests. So the retransmit setting is a left over from the tests. With the test i wanted to find a way to start razor-check and pyzor check without a systemd unit (for BSDs etc). So i ended up using this socat startup:
```
socat -v -t 10 TCP-LISTEN:5953,bind=127.0.0.1,fork EXEC:"/usr/bin/pyzor check"
```
and for razor
```
socat -v -t 10 TCP-LISTEN:11342,bind=127.0.0.1,fork SYSTEM:"/usr/bin/razor-check && printf spam || printf ham"
```
